share/completions/exif.fish

```diff
@@ -0,0 +1,89 @@
+function __get_target_files
+  set param_args (string split ', ' -- (string trim (string replace -fr "=\w+.*" '' -- (exif --help))))
+  set tokens (commandline -po)[2..]
+  set current_token 1
+  while test "$current_token" -le (count $tokens)
+    if contains -- $tokens[$current_token] $param_args
+      set -e tokens[$current_token]
+      set -e tokens[$current_token]
+    else if string match -- '-*' $tokens[$current_token]
+      set -e tokens[$current_token]
+    else
+      set current_token (math $current_token + 1)
+    end
+  end
+  echo $tokens
+end
+
+function __get_target_files_tags
+  string trim (string split -m1 ' ' (string replace -r '(\s+[-*]){5}' '' (string replace -fr '^(?!EXIF)' '' (exif --list-tags (__get_target_files)))))
+end
+
+function __get_potential_targets
+  set matching_files (commandline -t)*
+  for file in $matching_files
+    if test -d "$file"
+      echo "$file/"
+    else if exif "$file" &> /dev/null
+      echo "$file"
+    end
+  end
+end
+
+function __token_begins_with_arg
+  not string match -- '-*' (commandline -t)
+end
+
+complete -c exif -f -a "(__get_potential_targets)" -n "__token_begins_with_arg"
+
+for line in (exif --help)
+  if string match -r '^\s+-[\w\|\?],\s--(\w|-|=)+\s+.*$' "$line"
+    set parts (string split ', ' -- (string trim "$line"))
+    set short (string replace -r '^-' '' -- "$parts[1]")
+    set parts (string split -nm 1 ' ' -- "$parts[2]")
+  else if string match -r '^\s+--(\w|-|=)+\s+.*$' "$line"
+    set parts (string split -nm 1 ' ' -- (string trim "$line"))
+  else
+    continue
+  end
+
+  set long (string replace -r '^--' '' -- "$parts[1]")
+  set description (string trim "$parts[2]")
+
+  if string match -r '[\w-]+=[\w]' "$long"
+    set sub_parts (string split -nm 1 '=' "$long")
+    set long "$sub_parts[1]"
+    switch (string lower "$sub_parts[2]")
+      case "tag"
+        if test -n "$short"
+          complete -c exif -s "$short" -l "$long" -d "$description" -x -a "(__get_target_files_tags)"
```

Comment:
This just generates the completions. Honestly, typically I prefer hand-rolled completions because it allows you to tweak the descriptions.

You could just use the output of this and improve it, instead of running this all the time with a bunch of regexes that break the next time `exif` switches its output formatting.

Reply:
I suppose it's not super pretty, but from what I can see from the source code, `exif` uses the [popt library](https://github.com/devzero2000/POPT), so I don't think the output is likely to change formatting, and this method should survive changes to exif's functionality after the change in 936c9847 to handle miscellaneous required argument names.